pkg/util/trivy/trivy.go

```diff
@@ -407,8 +407,8 @@ func (c *Collector) ScanFilesystem(ctx context.Context, fsys fs.FS, path string,
 	return c.scanFilesystem(ctx, fsys, path, nil, scanOptions)
 }
 
-func (c *Collector) scan(ctx context.Context, artifact artifact.Artifact, applier applier.Applier, imgMeta *workloadmeta.ContainerImageMetadata, cache CacheWithCleaner) (*types.Report, error) {
-	if imgMeta != nil && cache != nil {
+func (c *Collector) scan(ctx context.Context, artifact artifact.Artifact, applier applier.Applier, imgMeta *workloadmeta.ContainerImageMetadata, cache CacheWithCleaner, useCache bool) (*types.Report, error) {
+	if useCache && imgMeta != nil && cache != nil {
```

Comment:
This code path is only used to populate the cache when layers are scanned one by one. When scanning from fs, the cache key can be computed only after doing the work so it does not save any cost. Even worse, we do the work twice (once in this code path and once in the scan function)